<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>MiXCR &mdash; Using AIRR-C Reference Sets 0.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=92fd9be5" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=837179f8"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="10X" href="10x.html" />
    <link rel="prev" title="Change-O" href="immcantation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Using AIRR-C Reference Sets
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="ogrdb.html">OGRDB REST API</a></li>
<li class="toctree-l1"><a class="reference internal" href="igblast.html">IgBLAST stand-alone</a></li>
<li class="toctree-l1"><a class="reference internal" href="immcantation.html">Change-O</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">MiXCR</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#introduction">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#downloading-and-processing-airr-c-reference-sets">Downloading and processing AIRR-C Reference Sets</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#test">Test</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="10x.html">10X</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Using AIRR-C Reference Sets</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">MiXCR</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/pages/mixcr.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="mixcr">
<h1>MiXCR<a class="headerlink" href="#mixcr" title="Link to this heading"></a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Link to this heading"></a></h2>
<p>MiXCR is an extremely widely-used tool for IG/TR analysis - MiXCR has its own references,
which are distributed with the program.</p>
<p>MiXCR has many presets and options for specifying how alignment is performed for different library preparation and
sequencing methods. This tutorial reflects one example - Takara’s SMART-seq for BCR with UMIs.</p>
</section>
<section id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Link to this heading"></a></h2>
<p>Mixcr installation requires that you have a license. Please refer to their website
for all details on installation. Briefly:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://github.com/milaboratory/mixcr/releases/download/v4.6.0/mixcr-4.6.0.zip
unzip<span class="w"> </span>mixcr-4.6.0.zip
<span class="c1">## Move to binary path or add directory to your path</span>
<span class="nb">export</span><span class="w"> </span><span class="nv">PATH</span><span class="o">=</span><span class="nv">$PATH</span>:<span class="nv">$PWD</span>
mixcr<span class="w"> </span>activate-license
<span class="c1">## Enter your key</span>
</pre></div>
</div>
</section>
<section id="downloading-and-processing-airr-c-reference-sets">
<h2>Downloading and processing AIRR-C Reference Sets<a class="headerlink" href="#downloading-and-processing-airr-c-reference-sets" title="Link to this heading"></a></h2>
<p>The following code block demonstrates how to download the human AIRR-C Reference Sets.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>airrc_refs
curl<span class="w"> </span>https://ogrdb.airr-community.org/api/germline/set/Human/IGH_VDJ/published/ungapped_ex<span class="w"> </span>&gt;<span class="w"> </span>airrc_refs/human_VDJ.fasta
curl<span class="w"> </span>https://ogrdb.airr-community.org/api/germline/set/Human/IGKappa_VJ/published/ungapped_ex<span class="w"> </span>&gt;&gt;<span class="w"> </span>airrc_refs/human_IGKVJ.fasta
curl<span class="w"> </span>https://ogrdb.airr-community.org/api/germline/set/Human/IGLambda_VJ/published/ungapped_ex<span class="w"> </span>&gt;&gt;<span class="w"> </span>airrc_refs/human_IGLVJ.fasta
</pre></div>
</div>
<p>The following code block is an example of how one can extract the V, D and J genes individually from the resultant FASTA files
(pip install Biopython first).</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>

<span class="n">seqs</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">((</span><span class="n">p</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span> <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="s1">&#39;airrc_refs/human_VDJ.fasta&#39;</span><span class="p">,</span> <span class="s1">&#39;fasta&#39;</span><span class="p">))</span>
<span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;V&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="s1">&#39;J&#39;</span><span class="p">]:</span>
  <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;airrc_refs/</span><span class="si">{</span><span class="n">gene</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span><span class="si">}</span><span class="s1">-genes.IGH.fasta&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">k</span><span class="p">:</span>
    <span class="p">[</span><span class="n">k</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;&gt;</span><span class="si">{</span><span class="n">seq_id</span><span class="si">}</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">+</span> <span class="n">seqs</span><span class="p">[</span><span class="n">seq_id</span><span class="p">]</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">seq_id</span> <span class="ow">in</span> <span class="n">seqs</span> <span class="k">if</span> <span class="n">seq_id</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span> <span class="o">==</span> <span class="n">gene</span><span class="p">]</span>

<span class="n">seqs</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">((</span><span class="n">p</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span> <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="s1">&#39;airrc_refs/human_IGKVJ.fasta&#39;</span><span class="p">,</span> <span class="s1">&#39;fasta&#39;</span><span class="p">))</span>
<span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;V&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="s1">&#39;J&#39;</span><span class="p">]:</span>
  <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;airrc_refs/</span><span class="si">{</span><span class="n">gene</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span><span class="si">}</span><span class="s1">-genes.IGK.fasta&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">k</span><span class="p">:</span>
    <span class="p">[</span><span class="n">k</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;&gt;</span><span class="si">{</span><span class="n">seq_id</span><span class="si">}</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">+</span> <span class="n">seqs</span><span class="p">[</span><span class="n">seq_id</span><span class="p">]</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">seq_id</span> <span class="ow">in</span> <span class="n">seqs</span> <span class="k">if</span> <span class="n">seq_id</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span> <span class="o">==</span> <span class="n">gene</span><span class="p">]</span>

<span class="n">seqs</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">((</span><span class="n">p</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span> <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="s1">&#39;airrc_refs/human_IGLVJ.fasta&#39;</span><span class="p">,</span> <span class="s1">&#39;fasta&#39;</span><span class="p">))</span>
<span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;V&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="s1">&#39;J&#39;</span><span class="p">]:</span>
  <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;airrc_refs/</span><span class="si">{</span><span class="n">gene</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span><span class="si">}</span><span class="s1">-genes.IGL.fasta&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">k</span><span class="p">:</span>
    <span class="p">[</span><span class="n">k</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;&gt;</span><span class="si">{</span><span class="n">seq_id</span><span class="si">}</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">+</span> <span class="n">seqs</span><span class="p">[</span><span class="n">seq_id</span><span class="p">]</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">seq_id</span> <span class="ow">in</span> <span class="n">seqs</span> <span class="k">if</span> <span class="n">seq_id</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span> <span class="o">==</span> <span class="n">gene</span><span class="p">]</span>
</pre></div>
</div>
<p>MiXCR has a nice functionality for building a custom reference or “library”. Again, please refer to MiXCR’s documentation and the specific protocol you used
to verify that this is optimal in your case.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mixcr<span class="w"> </span>buildLibrary<span class="w"> </span>--debug<span class="w"> </span>--v-genes-from-fasta<span class="w"> </span>airrc_refs/v-genes.IGH.fasta<span class="w"> </span>--v-gene-feature<span class="w"> </span>VRegion<span class="w"> </span>--j-genes-from-fasta<span class="w"> </span>airrc_refs/j-genes.IGH.fasta<span class="w"> </span>--d-genes-from-fasta<span class="w"> </span>airrc_refs/d-genes.IGH.fasta<span class="w"> </span>--c-genes-from-species<span class="w"> </span>human<span class="w"> </span>--chain<span class="w"> </span>IGH<span class="w"> </span>--taxon-id<span class="w"> </span><span class="m">9606</span><span class="w"> </span>--species<span class="w"> </span>human<span class="w"> </span>airrc-IGH.json.gz<span class="w"> </span>-f
mixcr<span class="w"> </span>buildLibrary<span class="w"> </span>--debug<span class="w"> </span>--v-genes-from-fasta<span class="w"> </span>airrc_refs/v-genes.IGK.fasta<span class="w"> </span>--v-gene-feature<span class="w"> </span>VRegion<span class="w"> </span>--j-genes-from-fasta<span class="w"> </span>airrc_refs/j-genes.IGK.fasta<span class="w"> </span>--c-genes-from-species<span class="w"> </span>human<span class="w"> </span>--chain<span class="w"> </span>IGK<span class="w"> </span>--taxon-id<span class="w"> </span><span class="m">9606</span><span class="w"> </span>--species<span class="w"> </span>human<span class="w"> </span>airrc-IGK.json.gz<span class="w"> </span>-f
mixcr<span class="w"> </span>buildLibrary<span class="w"> </span>--debug<span class="w"> </span>--v-genes-from-fasta<span class="w"> </span>airrc_refs/v-genes.IGL.fasta<span class="w"> </span>--v-gene-feature<span class="w"> </span>VRegion<span class="w"> </span>--j-genes-from-fasta<span class="w"> </span>airrc_refs/j-genes.IGL.fasta<span class="w"> </span>--c-genes-from-species<span class="w"> </span>human<span class="w"> </span>--chain<span class="w"> </span>IGL<span class="w"> </span>--taxon-id<span class="w"> </span><span class="m">9606</span><span class="w"> </span>--species<span class="w"> </span>human<span class="w"> </span>airrc-IGL.json.gz<span class="w"> </span>-f
mixcr<span class="w"> </span>mergeLibrary<span class="w"> </span>airrc-IGH.json.gz<span class="w"> </span>airrc-IGK.json.gz<span class="w"> </span>airrc-IGL.json.gz<span class="w"> </span>airrc.json.gz
</pre></div>
</div>
<section id="test">
<h3>Test<a class="headerlink" href="#test" title="Link to this heading"></a></h3>
<p>To test your installation, we’ll use example data as referenced by MiXCR:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-nc<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952573/ERR5952573_1.fastq.gz
wget<span class="w"> </span>-nc<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952573/ERR5952573_2.fastq.gz
</pre></div>
</div>
<p>This would be their default using the Takara preset:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mixcr<span class="w"> </span>analyze<span class="w"> </span>takara-human-rna-bcr-umi-smartseq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>ERR5952573_1.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>ERR5952573_2.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>results_default
</pre></div>
</div>
<p>To use our custom reference set, we have to split up the nice “analyze” preset into its constituent
parts:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mixcr<span class="w"> </span>align<span class="w"> </span>--preset<span class="w"> </span>takara-human-rna-bcr-umi-smartseq<span class="w"> </span>ERR5952573_1.fastq.gz<span class="w"> </span><span class="se">\</span>
ERR5952573_2.fastq.gz<span class="w"> </span>airrc_alignments.vdjca<span class="w"> </span>--species<span class="w"> </span>human<span class="w"> </span>--library<span class="w"> </span>airrc<span class="w"> </span>-f

mixcr<span class="w"> </span>refineTagsAndSort<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>airrc_alignments.vdjca<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>airrc_alignments.refined.vdjca

mixcr<span class="w"> </span>assemble<span class="w"> </span>airrc_alignments.refined.vdjca<span class="w"> </span>airrc_clones.clns
</pre></div>
</div>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="immcantation.html" class="btn btn-neutral float-left" title="Change-O" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="10x.html" class="btn btn-neutral float-right" title="10X" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Eve Richardson, William Lees.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>